src/evict/evict_page.c

```diff
@@ -680,7 +681,8 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
              * started between previous check and acquiring snapshot. If there is a checkpoint
              * running, release the snapshot and fallback to global visibility checks.
              */
-            if (conn->txn_global.checkpoint_running) {
+            checkpoint_running = conn->txn_global.checkpoint_running;
```

Comment:
Isn't there a risk of gcc optimizing this out if we don't use a barrier?

Reply:
This is in the same thread. The first read must happen before this read.

Reply:
I don't think gcc can do this.